### PR TITLE
feat(api-hub): adjust ms build to generate swagger file 

### PR DESCRIPTION
**Pull request by Phil91**
## Description

* update the swagger nuget package to be able to create the swagger documentation while building the project
* adjust ms build processes to generate an openApi file and copy it to the docs/api folder

## Why

To align with [TRG 1.08](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-08/)

## Issue

Refs: #1021

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml [hunk 1]
```diff
@@ -0,0 +1,7819 @@
+openapi: 3.0.1
+info:
+  title: Org.Eclipse.TractusX.Portal.Backend.Administration.Service
+  version: v2.3.0
```
Comment (evegufy) [inline docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml:4]: the version suffix isn't considered: https://github.com/eclipse-tractusx/portal-backend/blob/feature/1021-api-hub/src/Directory.Build.props#L23
could you please adjust this?
(applies to all services)

**Review by evegufy — CHANGES_REQUESTED** [review]

File: docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml (hunk 1 above)
Comment (Phil91) [reply docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml]: it's adjusted :) actually the assembly version don't include the prefixes.

File: .tractusx [hunk 2]
```diff
@@ -18,3 +18,9 @@
 ###############################################################
 
 leadingRepository: "https://github.com/eclipse-tractusx/portal"
+openApiSpecs:
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.2.0/docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.2.0/docs/api/Org.Eclipse.TractusX.Portal.Backend.Apps.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.2.0/docs/api/Org.Eclipse.TractusX.Portal.Backend.Notifications.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.2.0/docs/api/Org.Eclipse.TractusX.Portal.Backend.Registration.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.2.0/docs/api/Org.Eclipse.TractusX.Portal.Backend.Services.Service.yaml"
```
Comment (evegufy) [inline .tractusx:26]: Please change to next/upcoming version:
```suggestion
openApiSpecs:
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Apps.Service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Notifications.Service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Registration.Service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Services.Service.yaml"
```

**Review by evegufy — CHANGES_REQUESTED** [review]
Build is failing, maybe due to the implementation of the skip config validation

File: .tractusx [hunk 3]
```diff
@@ -18,3 +18,9 @@
 ###############################################################
 
 leadingRepository: "https://github.com/eclipse-tractusx/portal"
+openApiSpecs:
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Administration.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Apps.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Notifications.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Registration.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/Org.Eclipse.TractusX.Portal.Backend.Services.Service.yaml"
```
Comment (evegufy) [inline .tractusx:26]:
```suggestion
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/administration-service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/apps-service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/notifications-service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/registration-service.yaml"
- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/services-service.yaml"
```

Comment (Phil91) [reply .tractusx]: did do the rebase and adjusted the links

File: .tractusx [hunk 4]
```diff
@@ -19,8 +19,8 @@
...
-- "https://raw.githubusercontent.com/eclipse-tractusx/portal-backend/752ee7abf52606402d4053fe6edb9b8a05a74b84/docs/api/Org.Eclipse.TractusX.Portal.Backend.Notifications.Service.yaml"
-- "https://raw.githubusercontent.com/eclipse-tractusx/portal-backend/752ee7abf52606402d4053fe6edb9b8a05a74b84/docs/api/Org.Eclipse.TractusX.Portal.Backend.Registration.Service.yaml"
-- "https://raw.githubusercontent.com/eclipse-tractusx/portal-backend/752ee7abf52606402d4053fe6edb9b8a05a74b84/docs/api/Org.Eclipse.TractusX.Portal.Backend.Services.Service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/administration-service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/apps-service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/notifications-service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/registration-service.yaml"
+- "https://github.com/eclipse-tractusx/portal-backend/tree/v2.3.0-alpha.2/docs/api/services-service.yaml"
```
[the remark this answers is not in the document]
Comment (Phil91) [reply .tractusx]: adjusted

**Review by evegufy — DISMISSED** [review]

**Review by ntruchsess — APPROVED** [review]

**Review by ntruchsess — APPROVED** [review]

**Review by evegufy — APPROVED** [review]

**Pull request merged**